include/clang/StaticAnalyzer/Core/BugReporter/InternalInfo.h

```diff
@@ -93,6 +93,82 @@ class InternalInfoMap : private std::map<InternalInfo::Kind, std::unique_ptr<Int
 #endif
 };
 
+class ConstraintInfo : public InternalInfo {
+public:
+  using ConstraintMap = std::map<llvm::SmallString<200>, llvm::SmallString<200>>;
```

Comment:
Why not an `unordered_map`? Is the order important?

Reply:
LLVM has a high performance data structure for maps of strings, called StringMap http://llvm.org/doxygen/classllvm_1_1StringMap.html